python/foglamp/plugins/south/http_south/http_south.py

```diff
@@ -168,37 +181,35 @@ def plugin_reconfigure(handle, new_config):
         new_handle: new handle to be used in the future calls
     Raises:
     """
+    global loop
     _LOGGER.info("Old config for HTTP south plugin {} \n new config {}".format(handle, new_config))
 
-    # Find diff between old config and new config
-    diff = utils.get_diff(handle, new_config)
+    # plugin_shutdown
+    plugin_shutdown(handle)
+
+    # plugin_init
+    new_handle = plugin_init(new_config)
+
+    # plugin_start
+    plugin_start(new_handle)
 
-    # Plugin should re-initialize and restart if key configuration is changed
```

Comment:
May be shutdown & init plugin only if one of the key param is changing. E.g. when asset name is being changed, plugin shutdown and init is unnecessary.

Reply:
http_south server does not have any asset name per se, it just adds a prefix. Besides, it is run via a future which gets initialized once at start. If we want to change it, we will need to cancel it and then restart it. Hence this arrangement.